src/main/java/de/tum/in/www1/artemis/service/scheduled/DataExportScheduleService.java

```diff
@@ -82,43 +69,20 @@ private void checkSecurityUtils() {
         }
     }
 
-    /**
-     * Schedule the creation of a single data export
-     *
-     * @param dataExport the data export to be created
-     */
-    public void scheduleDataExportCreation(DataExport dataExport) {
-        log.info("Scheduling data export for {}", dataExport.getUser().getLogin());
-        scheduler.schedule(createDataExport(dataExport), retrieveScheduledDateTime());
-
-    }
-
     /**
      * Schedule the deletion of a single data export
      *
      * @param dataExport the data export to be deleted
      */
-    public void scheduleDataExportDeletion(DataExport dataExport) {
+    private void scheduleDataExportDeletion(DataExport dataExport) {
```

Comment:
This now only works if the Artemis server keeps running continuously for the whole week.

Is the creation date stored in the database (or something similar where you can tell how old it is)? Then you could at the end of `createDataExports()` make two additional calls (pseudocode)
```java
var dataExports = dataExportRepository.findAllThatNeedToBeDeleted();
dataExports.forEach(this::deleteDataExport);
```

Reply:
or alternatively: Add a second cron-scheduled method for that task so that the two things are handled in parallel and independent of each other.

(That should be fine without possible race-condition since the can be no overlap in the two groups of to be created and to be deleted, right?)